content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-state-2.md

```diff
@@ -0,0 +1,210 @@
+---
+date: '2021-09-15'
+section_menu: corda-5-dev-preview
+menu:
+  corda-5-dev-preview:
+    identifier: corda-corda-5.0-dev-preview-1-tutorial-c5-basic-cordapp-state-2
+    parent: corda-5-dev-preview-1-tutorials-building-cordapp
+    weight: 1030
+tags:
+- tutorial
+- cordapp
+title: "Write an advanced state"
+---
+
+Now that you have created a state to represent a voucher, you must create a `BoardingTicket` state to represent the ticket that the customer receives when they redeem their `MarsVoucher`. This state has some additional complexity. You will be creating this state in the same directory as `MarsVoucher` (`contracts/src/main/kotlin/net/corda/missionMars/states/`).
```

Comment:
Why do we say "this state has some additiona complexity" and not say what shape that complexity will take?

Reply:
Let's talk about this one in a call. I need a little help here.